birdy/client/base.py

```diff
@@ -180,8 +202,8 @@ def _method_factory(self, pid):
         func_builder = FunctionBuilder(
             name=sanitize(pid),
             doc=utils.build_process_doc(process),
-            args=["self"] + list(input_defaults),
-            defaults=tuple(input_defaults.values()),
+            args=["self"] + input_names,
+            defaults=defaults,
```

Comment:
Shouldn't `defaults` be a dict?

Reply:
a dict with key=input_identifer, value=default.

Reply:
The boltons documentation is actually wrong on this... I submitted a PR: https://github.com/mahmoud/boltons/pull/212
The `defaults` argument is passed to the `__defaults__` special argument, which expects a tuple.

Reply:
Ok.